babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -788,69 +788,56 @@ SqlIndex SqlCreateTableIndex(Span s) :
 SqlNode SqlExecMacro() :
 {
     SqlIdentifier macro;
-    SqlNodeList paramNames = new SqlNodeList(getPos());
-    SqlNodeList paramValues = new SqlNodeList(getPos());
+    SqlNodeList params = new SqlNodeList(getPos());
     Span s;
 }
 {
     macro = CompoundIdentifier() { s = span(); }
     [
-        SqlExecMacroArgument(paramNames, paramValues)
+        SqlExecMacroArgument(params)
     ]
     {
-        return new SqlExecMacro(s.end(this), macro, paramNames, paramValues);
+        return new SqlExecMacro(s.end(this), macro, params);
     }
 }
 
-void SqlExecMacroArgument(SqlNodeList paramNames, SqlNodeList paramValues) :
+void SqlExecMacroArgument(SqlNodeList params) :
```

Comment:
It seems like this function could be simplified by removing the "|" and having the following be optional in the top clause:

[
  name = SimpleIdentifier()
  "EQ"
]

And then a single constructor could be used, with name passed in optionally as null.

Reply:
Furthermore, I agree with @dstekol  that this pattern could be extracted into a helper parsing function so as to remove as much duplication as possible from the initial case and the case within ()*.

Reply:
Regarding your first comment, if we simplified the pattern to the way you suggested, we will be allowing mix and match for parameter with and without name to be parsed in the same list. Which is not the desired behavior. My thought is not to delay the validation till runtime. Any suggestions?

Reply:
Regarding your second comment, since two of you all pointed out extracting the pattern would be preferable. I am going to extract the pattern.

Reply:
That's a good point that blending the two would allow mix-and-match parameters, which we don't want.

Reply:
Pattern extracted in https://github.com/googleinterns/calcite/pull/108/commits/823f91878be465524acca95e35c28db65f20a970